### DISCUSSION TITLE
Error related to openpose

### DISCUSSION THREAD

**Discussion by chris19299** [Mikubill/sd-webui-controlnet#481] [General]
File "C:\Users\chris\stable-diffusion-webui-directml\extensions\sd-webui-controlnet\annotator\openpose\body.py", line 57, in __call__
    heatmap = cv2.resize(heatmap, (0, 0), fx=stride, fy=stride, interpolation=cv2.INTER_CUBIC)
cv2.error: OpenCV(4.7.0) :-1: error: (-5:Bad argument) in function 'resize'
> Overload resolution failed:
>  - src data type = 23 is not supported
>  - Expected Ptr<cv::UMat> for argument 'src'

Is that the problem of using AMD GPU ?

**Comment by AugmentedRealityCat** [comment]
The error message points to the annotator folder - so this is likely a problem with the annotator, the first step of the process where a normal picture is analyzed to estimate the pose of all the people on it and encode that in a colored-bones-over-black-background picture, and not the second step, where that colored-bones-over-black-background picture is interpreted by Stable Diffusion via either ControlNet or T2i (both now support Openpose).

Can you check if it works by not using the annotator and feeding the ControlNet / T2i directly with a pre-processed image in the proper format, with colored bones over a black background ?

And to help debug the annotator part, can you check what is in that `stable-diffusion-webui-directml\extensions\sd-webui-controlnet\annotator\openpose\` folder ? In particular, can you check if you have the `body_pose_model.pth` file there ?

**Comment by chris19299** [comment]
I change the code to be 
heatmap = cv2.resize(heatmap.astype('float32'), (0, 0), fx=stride, fy=stride, interpolation=cv2.INTER_CUBIC)

Then the problem fixed.
